R/utils.R

```diff
@@ -947,15 +949,21 @@ columnToRowData <- function(data) {
 #' shinyApp(ui, server)
 #' }
 #' @export
-customStop <- function(error, errorClass = character(0)) {
+safeError <- function(error, errorClass = character(0)) {
   if (inherits(error, "error")) {
-    class(error) <- c("shiny.custom.error", errorClass, class(error))
-    stop(error)
+    cond <- structure(
+      list(message = error$message),
+      class = c("shiny.custom.error", errorClass, class(error))
+    )
   } else if (inherits(error, "character")) {
-    stopWithCondition(c("shiny.custom.error", errorClass), error)
+    cond <- structure(
+      list(message = error),
+      class = c("shiny.custom.error", errorClass, "error", "condition")
+    )
```

Comment:
I was actually picturing more like this (not tested):

```r
if (inherits(error, "character")) {
  error <- simpleError(error)
}
if (!inherits(error, "error")) {
  stop("The class of the `error` parameter must be either 'error' or 'character'")
}
if (!("shiny.custom.error" %in% class(error))
  class(error) <- c("shiny.custom.error", class(error))
error
```

I'd like to use `simpleError` to convert the error string to an error obj, as it relies less on the internal implementation details of errors. And for the case where error is an error obj already, we can leave it alone as much as possible and just add the one class name.